### DISCUSSION TITLE
Hosting and AOT trimm?

### DISCUSSION THREAD

**Discussion by RobertK66** [Tapanila/SharpCaster#301] [Ideas]
I do not know how many time you have to invest in some more features, but my thoughts on next steps are:

1. Make (a separate ? package for)  hosting usage available:
Not sure how to do this - i never did this - but I think it shouldn't be that complicated. The goal would be to have smt. like:
```
 IHostBuilder host = Host.CreateDefaultBuilder(args)
                         .UseSharpcaster(c=>{ c.UseAppId("xyz");
                                              c.UseReceiver( r => r.Name.Equals("My Receiver")); });
```
and if you only use UseSharpcaster() you get some default AppId ( is there one, one can use legally here ???) and take the first receiver found. (Or maybe to separate out the search and use only IP/port on that level of hosting the ChromCastClient() only !?

2. Make the library AOT ready.
That's completely new to me also. But if I try to set PublishTrimmed to true on my ConGui i do get a lot of [IL2026-Warnings](https://learn.microsoft.com/en-us/dotnet/core/deploying/trimming/trim-warnings/il2026). Lot of them from my own Logging and Hosting (...) but I think at least all the Messages and model classes should somehow be marked as 'do not trimm out if not referenced'. No idea how much of an effort this will be but I am willing to try and give some more input on that one if we want to go down that rabbit hole 😄 ....

**Comment by Tapanila** [comment]
The AOT part absolutely make sense. Why you would like to have the hosting part?

**Comment by RobertK66** [comment]
I am not so sure about hosting yet, but I really got used to (love it now)  when playing around with DI, Logging and Configuration in 'modern net8 scenarios'.  e.g. when using [Microsofts Aspire](https://learn.microsoft.com/en-us/dotnet/aspire/get-started/aspire-overview) together with [Microsoft Orleans](https://learn.microsoft.com/en-us/dotnet/orleans/overview) one can wire up multiple containerizable microservices for 'local distributed debugging' with following 'main':

```
using k8s.KubeConfigModels;
using Microsoft.Identity.Client.Extensions.Msal;

var builder = DistributedApplication.CreateBuilder(args);

var dbserver = builder.AddPostgres("dbserver")
                      .WithBindMount("../AspireApp2.ApiService/script",     // All Sh and sql found here are executed when container is created
                                     "/docker-entrypoint-initdb.d")
                      .WithPgAdmin();                       // This is the (Web)GUI to controll the DB server.

var catalogDb = dbserver.AddDatabase("catalogdb");          // 


var clusteringTable = builder.AddRedis("clustering");
var grainStorage = builder.AddRedis("grain-state").WithRedisCommander();

// Add the Orleans resource to the Aspire DistributedApplication
// builder, then configure it with Azure Table Storage for clustering
// and Azure Blob Storage for grain storage.
var orleans = builder.AddOrleans("default")
                     .WithClustering(clusteringTable)
                     .WithGrainStorage("Default", grainStorage);


var apiService = builder.AddProject<Projects.AspireApp2_ApiService>("apiservice")
    .WithReference(orleans)
    .WithReference(catalogDb)
    .WithReplicas(2);
    

builder.AddProject<Projects.AspireApp2_Web>("webfrontend")
    .WithExternalHttpEndpoints()
    .WithReference(orleans.AsClient())
    .WithReference(apiService);

builder.Build().Run();
```

Starting this gives you a webview with access to all (local) running parts, their logging and instrumentation, and all Projects added from your solution are running in debug mode fully connected to VS ....

![image](https://github.com/user-attachments/assets/1b7b37d0-81dd-46f8-95c4-be7b52107fd4)

**Reply by Tapanila** [reply]
This would make a lot more sense if Sharpcaster would have a REST API.

If you or anyone else has interest to build one I would be more than happy to add it to the project

**Comment by RobertK66** [comment]
give me some days to consider this, ....
